crates/core/src/json_pointer.rs

```diff
@@ -65,14 +61,17 @@ impl JsonPointer {
     /// # Safety
     ///
     /// The input string *must* be a valid JSON pointer.
-    pub const unsafe fn new_unchecked(s: &[u8]) -> &Self {
+    pub const unsafe fn new_unchecked(s: &str) -> &Self {
         std::mem::transmute(s)
     }
 
-    pub const fn validate(bytes: &[u8]) -> bool {
-        if std::str::from_utf8(bytes).is_err() {
+    /// Confirms the validity of a string such that it may be safely used for
+    /// [`Self::new_unchecked`].
+    pub const fn validate(s: &str) -> bool {
```

Comment:
Please make sure this accepts a `&[u8]`, or decline it into `validate_bytes` and `validate_str`.

Reply:
Sure, I've split it into two.